iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -252,52 +265,33 @@ public override async Task<Message> ReceiveAsync(CancellationToken cancellationT
-            responseMessage.Headers.TryGetValues(HttpResponseHeader.ETag.ToString(), out IEnumerable<string> lockToken);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.EnqueuedTime, out IEnumerable<string> enqueuedTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.DeliveryCount, out IEnumerable<string> deliveryCountAsStr);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.ExpiryTimeUtc, out IEnumerable<string> expiryTime);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.CorrelationId, out IEnumerable<string> correlationId);
-            responseMessage.Headers.TryGetValues(CustomHeaderConstants.SequenceNumber, out IEnumerable<string> sequenceNumber);
-
             byte[] byteContent = await responseMessage.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
 
             Message message = byteContent != null
                 ? new Message(byteContent)
                 : new Message();
 
-            message.MessageId = messageId?.First();
-            message.LockToken = lockToken?.First().Trim('\"');
-
-            if (enqueuedTime != null)
+            // Read Http headers and map them to message system properties.
+            foreach (KeyValuePair<string, IEnumerable<string>> header in responseMessage.Headers)
             {
-                if (DateTime.TryParse(enqueuedTime.First(), out DateTime enqueuedTimeUtc))
-                {
-                    message.EnqueuedTimeUtc = enqueuedTimeUtc;
-                }
-            }
+                string headerKey = header.Key;
 
-            if (deliveryCountAsStr != null)
-            {
-                if (byte.TryParse(deliveryCountAsStr.First(), out byte deliveryCount))
+                if (header.Key.Equals(HttpResponseHeader.ETag.ToString(), StringComparison.OrdinalIgnoreCase))
                 {
-                    message.DeliveryCount = deliveryCount;
+                    message.SystemProperties[MessageSystemPropertyNames.LockToken] = header.Value?.First().Trim('\"');
                 }
-            }
 
-            if (expiryTime != null)
-            {
-                if (DateTime.TryParse(expiryTime.First(), out DateTime absoluteExpiryTime))
+                if (s_mapHttpHeadersToMessageProperties.ContainsKey(headerKey))
                 {
-                    message.ExpiryTimeUtc = absoluteExpiryTime;
+                    string messagePropertyKey = s_mapHttpHeadersToMessageProperties[headerKey];
+                    object messagePropertyValue = ConvertToMessageSystemProperty(messagePropertyKey, header.Value);
+                    if (messagePropertyValue != null)
+                    {
+                        message.SystemProperties[messagePropertyKey] = messagePropertyValue;
+                    }
                 }
             }
 
-            message.CorrelationId = correlationId?.First();
-            message.SequenceNumber = sequenceNumber == null
```

Comment:
What are these properties? After the fix, these would be null right? Should we keep the values in these fields and the SystemProperties for backward compatibility?

Reply:
`ConvertToMessageSystemProperty` will initialize it with the value, if available, else set it to null. This is the existing behavior.